src/elements/TextInput.vue

```diff
@@ -78,47 +80,56 @@ const onInvalid = (evt: HTMLInputElementEvent) => {
  * the user does something worth invalidating
  */
 const onChange = () => {
+  inputRef.value.setCustomValidity('');
+  isInvalid.value = false;
   isDirty.value = true;
+  validationMessage.value = '';
 };
 </script>
 
 <template>
   <label class="wrapper" :for="name">
-    <span class="label">
+    <span v-if="$slots.default" class="label">
       <slot />
       <span v-if="required && !model?.length" class="required">*</span>
     </span>
     <span class="tbpro-input" :class="{ 'small-text': props.smallText }">
-      <span v-if="prefix" ref="inputPrefix" class="tbpro-input-prefix">{{
-        prefix
-      }}</span>
-      <input
-        class="tbpro-input-element"
-        v-model="model"
-        :class="{ dirty: isDirty }"
-        :type="type"
-        :id="name"
-        :name="name"
-        :disabled="disabled"
-        :placeholder="placeholder"
-        :required="required"
-        :maxLength="maxLength"
-        @invalid="onInvalid"
-        @change="onChange"
-        ref="inputRef"
-        :style="{ paddingLeft: inputPaddingLeft }"
-        :data-testid="dataTestid"
-      />
+      <span v-if="outerPrefix" class="tbpro-input-outer-prefix">{{ outerPrefix }}</span>
+      <span class="tbpro-input-wrapper">
```

Comment:
We need another wrapper element here to properly align the outer prefix with the rest of the component.